### DISCUSSION TITLE
Find out if device is playing

### DISCUSSION THREAD

**Discussion by daywink** [pschatzmann/ESP32-A2DP#335] [Q&A] [answered]
Hello, I have made a simple Bluetooth speaker with a touch display, and I want to implement a play/pause button.
Currently, the functionality does work, as I am using `get_audio_state()`, but after pausing and trying to play again, it is unable to until the audio state is back to Stopped or Suspended. Any way to fix this issue, and be sure that the audio is not playing?

**Comment by pschatzmann** [comment, marked as answer]
Hmm, I am not able to see any issue. To me it makes perfect sense that you can only push play() on something that is stopped or suspended and that you can stop() only something that is playing. I would suggest that you deactivate the buttons that are not valid at the actual state.
I would recommend to use the set_on_audio_state_changed() callback method so that you are informed about state changes.

**Reply by daywink** [reply]
Hi, I used the callback and it works good, but what I meant is the remote (e.g. iOS) can take a long time to suspend the audio state and it can be annoying. I made a workaround for this but it's not full-proof. 

Edit: For some reason `set_on_volumechange` isn't working for me. The code is as follows:
```
BTSPK.set_on_volumechange(volumechange); // This is the end of the setup() function
}

void volumechange(int volume) {
  Serial.println(volume);
  lv_slider_set_value(ui_BTSPKVolSlider, volume, LV_ANIM_ON);
}
```
No serial output, no display update.

**Reply by pschatzmann** [reply]
Just read the documentation: https://github.com/pschatzmann/ESP32-A2DP/wiki/Controlling-your-Phone-with-AVRC-Commands
